### PR TITLE
Consolidate release jobs

### PR DESCRIPTION
**Pull request by mdonnalley**
- Removes the `release-signed-package` job 
- Removes the `sign-and-publish` command
- Makes signing an option on the `release-package` job
- Adds `verify-signed-package` command
- Adds `sign` command
- `publish` command now waits for new dist tag to show up in npm registry

### CODE REVIEW

File: src/commands/publish.yml [hunk 1]
```diff
@@ -15,14 +15,51 @@ parameters:
   access:
     description: tells the registry whether this package should be published as public or restricted
     default: public
-    type: string
+    type: enum
+    enum: [public, restricted]
+  use_tarfile:
```
Comment (amphro) [inline src/commands/publish.yml:20]: I think I would rather just have this be `tarfile` that is a string. Then pass in `TARPATH` that is output from the trust plugin. That will protect against renames in the future and also make the command more flexible.

File: src/commands/sign.yml [hunk 2]
```diff
@@ -5,32 +5,24 @@
...
   - jq/install
+  - setup-aws
+  - install-sfdx-trust
   - run:
-      name: Set credentials
+      name: Set salesforce credentials
       command: |
-        echo //registry.npmjs.org/:_authToken=$NPM_TOKEN > /root/.npmrc
-        echo unsafe-perm = true >> /root/.npmrc
         echo $SALESFORCE_KEY | base64 --decode > /root/salesforce-cli.key
```
Comment (amphro) [inline src/commands/sign.yml:23]: See https://github.com/forcedotcom/lwc-dev-server/pull/246#discussion_r466769410

File: src/commands/publish.yml (hunk 1 above)
Comment (mdonnalley) [reply src/commands/publish.yml:20]: @amphro as far as I know there's no way to pass information from one command to the next, which is why I did it this way

Comment (amphro) [reply src/commands/publish.yml:20]: Dang, ok.

**Pull request merged**